src/cmakeTools.ts

```diff
@@ -574,11 +574,15 @@ export class CMakeTools implements api.CMakeToolsAPI {
                     const changeSourceDirectory = localize('edit.setting', "Locate");
                     const ignoreCMakeListsMissing = localize('ignore.activation', "Don't show again");
 
-                    let showCMakeLists: boolean = await expShowCMakeLists();
+                    let showCMakeLists: boolean = await showCMakeListsExperiment();
                     const existingCmakeListsFiles: string[] | undefined = await util.getAllCMakeListsPaths(this.folder.uri);
-                    if (showCMakeLists) {
-                        showCMakeLists = existingCmakeListsFiles !== undefined && existingCmakeListsFiles.length > 0;
-                        telemetryProperties["ignoreExperiment"] = (!showCMakeLists).toString();
+
+                    telemetryProperties["showCMakeListsExperiment"] = (showCMakeLists).toString();
+                    if (existingCmakeListsFiles !== undefined && existingCmakeListsFiles.length > 0) {
```

Comment:
A little bit simpler it can be done like this:
```ts
if (existingCmakeListsFiles === undefined || existingCmakeListsFiles.length == 0) {
    showCMakeLists = false;
}
telemetryProperties["hasCMakeLists"] = showCMakeLists.toString();
```

Reply:
@bobbrow showCMakeLists is always false for the control group, no matter whether there is a cmakelists.txt available or not.

Reply:
got it